internal/worker/worker.go

```diff
@@ -779,28 +783,32 @@ func (s *Worker) startup() error {
 }
 
 func (s *Worker) shutdown() error {
-	log.Printf("Worker spinning down.")
+	log.Println("Shutting down...")
 	defer s.eventRepo.PushWorkerStoppedEvent(s.workerId)
 
+	var errs error
+
 	worker, err := s.workerRepo.GetWorkerById(s.workerId)
 	if err != nil {
-		return err
+		errs = errors.Join(errs, err)
 	}
 
-	s.cancel()
-
 	err = s.workerRepo.RemoveWorker(worker)
```

Comment:
In the case `worker, err := s.workerRepo.GetWorkerById(s.workerId)` fails, worker can be `nil`. Since we are not immediately returning the error, there is a likelihood the program will panic at `err = s.workerRepo.RemoveWorker(worker)` in this block
`err := r.lock.Acquire(context.TODO(), common.RedisKeys.SchedulerWorkerLock(workerId), common.RedisLockOptions{TtlS: 10, Retries: 3})`

Reply:
Updated. Thoughts?

Reply:
I think this is good